consolidate_go_stacktrace: Add --filter lock

joestringer opened the pull request:
Add a filter for events + summary for packages involved in lock issues

joestringer committed:
consolidate_go_stacktrace: Add --filter lock

Add a filter for events + summary for packages involved in lock issues

Signed-off-by: Joe Stringer <joe@cilium.io>

michi-covalent reviewed (approved):
i know it's still a draft but i'm approving it anyways.

joestringer committed:
consolidate_go_stacktrace: Fix relative paths

Previously, relative paths just truncated the beginning of the line, eg.
pkg/foo instead of beginning with the relative path characters
./pkg/foo. Fix this by ensuring the package paths begin with "./" .

Fixes: bafbfb8452db ("consolidate_go_stacktrace.py: Use relative paths by default")
Signed-off-by: Joe Stringer <joe@cilium.io>
